src/common/services/api/order.service.js

```diff
@@ -316,6 +340,87 @@ class Order {
       return startedOrderWithoutSpouse
     }
   }
+
+  calculatePricesWithFees (feesApplied, cartItems) {
+    angular.forEach(cartItems, (item) => {
+      if (feesApplied) {
+        item.amountWithFee = item.amount
+      } else {
+        item.amountWithFee = this.calculatePriceWithFees(item.amount)
+      }
+    })
+    return true
+  }
+
+  calculatePriceWithFees (originalAmount) {
+    const newAmount = this.calculateAmountWithFees(originalAmount)
+    return this.$filter('number')(newAmount, 2)
+  }
+
+  calculateAmountWithFees (originalAmount) {
+    originalAmount = parseFloat(originalAmount)
+    return (originalAmount * 0.0235) +
+      (originalAmount * 0.0235 * 0.0235) +
+      (originalAmount * 0.0235 * 0.0235 * 0.0235) +
+      (originalAmount * 0.0235 * 0.0235 * 0.0235 * 0.0235) +
+      originalAmount
```

Comment:
What does this math represent?

Reply:
It's mentioned in the Jira ticket (linked in the description). However, the basic idea is that it is a 2.35% processing fee.

Reply:
Seems like there's got to be a better and more accurate way to calculate that. Does this work? Maybe there's something I don't have context for. Seems like you can get it down to a multiplication by a constant with you want. My example computations seem to match your document.

Equations:
```
amt_gift     = amt_with_fee - amt_with_fee * 0.0235
amt_gift     = (1 - 0.0235) * amt_with_fee
amt_with_fee = amt_gift / (1 - 0.0235)
amt_with_fee = amt_gift / 0.9765
amt_with_fee = amt_gift * (1 / 0.9765)
amt_with_fee = amt_gift * 1.0240655402
```
Examples:
```
     10.240 =      10 / (1 - 0.0235)
    102.406 =     100 / (1 - 0.0235)
  1,024.065 =   1,000 / (1 - 0.0235)
 10,240.655 =  10,000 / (1 - 0.0235)
102,406.554 = 100,000 / (1 - 0.0235)
```

Reply:
Hmm, not sure. I was given the calculation from the fund development team, it wasn't something I came up with.

Reply:
The numbers seem to come out the same way, so maybe that's fine.

Reply:
Maybe the goal is to avoid division, in which case I guess the approximation makes sense. If the precision of the approximation isn't enough, seems like Cru picks up the tab. But looks like you've got some pretty large gifts covered.

Reply:
Looks like it's slightly different at $1M:

```
1024065.5328550625
1024065.540194572452637 = 1,000,000 / (1 - 0.0235)
```

Reply:
Cuz the approximation runs out of precision and JS division still has precision... The approximation isn't increasing it enough to cover the whole fee. It's off by a whole cent...
`1024065.5328550625 - (1024065.5328550625 * .0235) = 999,999.992`

Reply:
Anyways this was a fun excursion in math but either solution is probably fine. I was just wondering why. Maybe some comment in the code about how the formula was provided by fund development would be good.

Reply:
I'll run your calculation by FDS and see what they think. They still have to do some more testing so there is time to update it if needed.

Reply:
Looks like `amt_gift     = amt_with_fee - amt_with_fee * 0.0235` is more precise than `amt_gift     = (1 - 0.0235) * amt_with_fee`. At 102,406.55 the first gives `100000.000075` and the latter gives `99999.996075`

Reply:
Must be a fun artifact of floating point number impentations. Cuz algebraically those should be equal.

Reply:
Ran the calculations in the console and they both came back with the right number, using the calculator app came with the wrong numbers.

Reply:
Good news, FDS approved the change in calculation, thanks Scotty!